Adding the "meta" tag and servos

madhephaestus opened the pull request:
In the Meta tag for each vitamin type there can be a "scriptGit" tag with a "scriptFile" tag that can take the given vitamin and make geometry out of it. The meta tag includes an option to flag as an actuator, an actuators shaft, or a fastener (none implement fastener yet).

madhephaestus committed:
adding servo

madhephaestus committed:
adding servo

madhephaestus committed:
fixing json

madhephaestus committed:
adding a script tag to the hobby servos

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
fixing nulled values

madhephaestus committed:
Update stepperMotor.json

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Update hobbyServo.json

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Cleaning up the directory to only being json

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
removing junk servo

madhephaestus committed:
gitignore

madhephaestus committed:
stripped down readme

madhephaestus committed:
chaging to meta tag

madhephaestus committed:
addign actuator tag

madhephaestus committed:
adding horn configurations

madhephaestus committed:
pushing horn configurations

madhephaestus committed:
fixed horn dimentions

madhephaestus committed:
adding microservo

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Update hobbyServo.json

madhephaestus committed:
adding a meta script

madhephaestus committed:
Update nut.json

madhephaestus committed:
Update ballBearing.json

madhephaestus committed:
Update ballBearing.json

madhephaestus committed:
Update linearBallBearing.json

madhephaestus committed:
Update timingBelt.json

madhephaestus committed:
Update stepperMotor.json

madhephaestus committed:
Update lockNut.json

madhephaestus commented:
I have added cad generators for all of the hardware in this repository. They each have a script that will generate a CSG object in bowler-script-kernel.

CSG vitamin = Vitamins.get("lockNut","M3")

To load vitamins from this library. The full API of listing options in code can be found:
https://gist.github.com/7c437ad8d943b6fa71d9c57fb2ab4df3.git

madhephaestus committed:
Update capScrew.json

madhephaestus committed:
Update ballBearing.json

madhephaestus committed:
Update linearBallBearing.json

madhephaestus committed:
Update lockNut.json

madhephaestus committed:
Update nut.json

madhephaestus committed:
Update timingBelt.json

sjkelly commented on json/capScrew.json:
```diff
@@ -1,4 +1,8 @@
 {
+  "meta": {
+    "scriptGit": "https://gist.github.com/cf5578c406644e7e4fe5447a698f85d5.git",
```
Can this be namespaced as bowler?

madhephaestus replied:
Sure, how would you like to see the schema?

sjkelly replied:
maybe something like:

``` json
"meta": {
  "bowler" :{
    "git": "foo",
    "tag": "bar"
  }
}
```

sjkelly commented:
The meta tags are a pretty good idea! Can you squash these commits though? Removing the ods should be fine, but I'd still like the csv files to remain.

sjkelly commented on README.md:
```diff
@@ -2,9 +2,9 @@
 
 ##Function
-Dimensions are stored in Open Document Spreadsheets (ods). CSV and JSON files generated by the build.sh script are available. LibreOffice is used to generate the CSV files.
+Dimensions are stored in  JSON files generated BowlerStudio. 
```
I think CSV -> JSON would be a bit nicer. What is Bowlerstudio generating here?

madhephaestus replied:
BowlerStudio is generating JSON from GSON and an internal datatype tuned to match the schema. Could we do a JSON ->CSV instead? JSON tends to be the "universal" format these days, and it seem bad to have more then one copy of "the data". Only one "copy" of the data should be the "true" data, and the other generated. i guess it comes down to the purpose, machine or human readable?

sjkelly replied:
I agree JSON is much better for the final application. However I like the ease at which you can do consistency checks with CSV, for example if a field is missing. Speaking of which, this may be good to add to a test suite. A test suite and just JSON may be sufficient to ensure everything is correct.

I prefer to edit the data in a spreadsheet. Are you providing forms to fill in the data?

sjkelly commented on json/stepperMotor.json:
```diff
@@ -1,52 +1,57 @@
-    "flangeDiameter": 21.0,
+    "mountScrew": "M3",
     "flangeHeight": 2.0,
-    "length": 27.0,
```
Can you make your JSON output sort by keys so these diffs are easier to review?

madhephaestus replied:
Id need to write a custom serializer to get it alphabetical. Maps are treated as unsorted data and stored based on some internal algorithm. This one is hard...

sjkelly replied:
There is one potential benefit of CSV -> JSON. This is how it is sorted currently: https://github.com/sjkelly/Hardware-Dimensions/blob/901e65dac42357b5c0178d6347abbec9775cfca4/csv_to_json.py#L28

madhephaestus committed:
Update ballBearing.json

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Pushing changed Database

madhephaestus committed:
Update nut.json

madhephaestus committed:
Update lockNut.json

madhephaestus committed:
Update nut.json

madhephaestus committed:
Update capScrew.json

madhephaestus committed:
Update hobbyServoHorn.json

CounterfeitLlama committed:
Added 6#32 screw

rpolhemus committed:
Added 6#32 bolt

Commit:
Added 6#32 screw + measurements

DataManiac18 committed:
Added 6#32 screw

used in vex robotics

Commit:
Added 6#32 cap screw

CounterfeitLlama committed:
Updated meta gist script

Commit:
Updated URL in meta

DataManiac18 committed:
Replaced meta scriptGit link

forked a new scriptGit repo and pasted mine in the meta data structure

Commit:
Update capScrew.json

rpolhemus committed:
Added own cap screw gen

madhephaestus committed:
Create vexFlatSheet.json

madhephaestus committed:
Update vexFlatSheet.json

madhephaestus committed:
Merge branch 'master' of git@github.com:NauticalTooth27/Hardware-Dimensions.git

madhephaestus committed:
Merge branch 'master' of git@github.com:myufus/Hardware-Dimensions.git

Conflicts:
	json/capScrew.json

madhephaestus committed:
Merge branch 'master' of git@github.com:pjmara/Hardware-Dimensions.git

Conflicts:
	json/capScrew.json

madhephaestus committed:
Merge branch 'master' of
git@github.com:DataManiac18/Hardware-Dimensions.git

Conflicts:
	json/capScrew.json

madhephaestus committed:
Merge branch 'master' of
git@github.com:CounterfeitLlama/Hardware-Dimensions.git

Conflicts:
	json/capScrew.json

DataManiac18 committed:
Merge pull request #1 from madhephaestus/master

This should be a pull request to myselft

CounterfeitLlama committed:
Merge pull request #1 from madhephaestus/master

Update personal branch from classwork

rpolhemus committed:
Merge pull request #1 from madhephaestus/master

Pulling from update to my branch

Commit:
Merge pull request #1 from madhephaestus/master

Updating MY hardware definitions

Commit:
Merge pull request #1 from madhephaestus/master

Pull request 10.4.2016

CounterfeitLlama committed:
Created vexGear.json

Commit:
Added all VEX standoff sizes

DataManiac18 committed:
Created the json file with vex wheel dimensions 

Started with the measurements of 4" standard, 2.75" smooth, and 4" smooth (high traction) wheels.

Commit:
Saved so I can close chrome without losing changes

DataManiac18 committed:
File can now compile

added a new "wheel" called "comment/future additions" with my plans for future additions.

DataManiac18 committed:
Fixed Parameter Name

rpolhemus committed:
Created VexSpacer.json, added initial measurements

DataManiac18 committed:
Fixed the Misspelling of Height

rpolhemus committed:
Fixed a stupid

"Spacer" read: "vexSpacer"

CounterfeitLlama committed:
Edited errors in diameter

Commit:
Put other file into correct folder

Commit:
changed to .json

madhephaestus committed:
Merge pull request #2 from NauticalTooth27/master

I am addinf Vex Spacers

madhephaestus committed:
Merge pull request #6 from myufus/master

Adding VEX standoff dimensions

madhephaestus committed:
Merge pull request #5 from DataManiac18/master

Adding Vex Wheels

madhephaestus committed:
Merge pull request #4 from pjmara/master

I am adding C-Channel wide and skinny

madhephaestus committed:
Merge pull request #3 from CounterfeitLlama/master

Adding Vex Gear

madhephaestus committed:
remove comments, this should be an Issue

madhephaestus committed:
Updating the names to remove invalid chars

madhephaestus committed:
moving script into main repo for editing

madhephaestus committed:
pointing to the mainline repo for script files

madhephaestus committed:
Pointing script at gear generator

madhephaestus committed:
Updating the linking of the vitamin to the source

madhephaestus committed:
Update vexStandoff.json

madhephaestus committed:
Adding sg90 horn

madhephaestus committed:
adding tower pro SG90

madhephaestus committed:
Update vexGear.json

madhephaestus committed:
Update hobbyServo.json

madhephaestus committed:
Create torsionSpring.json

madhephaestus committed:
Update torsionSpring.json

madhephaestus committed:
adding 360 torsion spring

Commit:
adding the 6 point horn

Commit:
adding comma

Commit:
pusing comma

Commit:
adding the commas back

madhephaestus committed:
adding larger bearing R8-2RS

madhephaestus committed:
Update ballBearing.json
